### PR TITLE
Fix non-base58 character error on windows

### PR DESCRIPTION
**Pull request by durac**
Replace backslash used in windows paths with slash to ensure correct conversion into parts.
With these changes orbit-db should support windows.

Related to #274

### CODE REVIEW

**Comment by CaptainQuark** [comment]
+1

**Comment by expressflow** [comment]
+1

**Comment by haadcode** [comment]
Thank you @durac for the PR! Sounds great if this would get us on Windows :) I'd like to try it myself on a Windows machine to see it all works and the developer experience is solid like it is osx/linux.

I'll comment on the PR if there's anything to comment.

File: package.json [hunk 1]
```diff
@@ -59,6 +59,13 @@
...
+    "test-win": "node node_modules\\mocha\\bin\\mocha",
+    "build-win": "npm run build:es5-win && npm run build:debug-win && npm run build:dist-win && npm run build:examples-win && npm run build:docs/toc-win",
+    "build:examples-win": "webpack --config conf/webpack.example.config.js --sort-modules-by size && if not exist examples\\browser\\lib mkdir examples\\browser\\lib && cp node_modules\\ipfs\\dist\\index.js examples\\browser\\lib\\ipfs.js",
+    "build:dist-win": "webpack --config conf\\webpack.config.js --sort-modules-by size && if not exist examples\\browser\\lib mkdir examples\\browser\\lib && cp dist\\orbitdb.min.js examples\\browser\\lib\\orbitdb.min.js",
+    "build:debug-win": "webpack --config conf\\webpack.debug.config.js --sort-modules-by size && if not exist examples\\browser\\lib mkdir examples\\browser\\lib && cp dist\\orbitdb.js examples\\browser\\lib\\orbitdb.js && cp dist\\orbitdb.js.map examples\\browser\\lib\\orbitdb.js.map",
+    "build:docs/toc-win": "markdown-toc --no-first1 -i README.md && markdown-toc --no-first1 -i API.md && markdown-toc --no-first1 -i GUIDE.md && markdown-toc --no-first1 -i CHANGELOG.md",
```
Comment (haadcode) [inline package.json]: This line seems to be the same as https://github.com/orbitdb/orbit-db/pull/453/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R61. Do we need a separate script for Windows or could we use the same scripts as above?

File: test/create-open.test.js [hunk 2]
```diff
@@ -138,7 +138,7 @@ Object.keys(testAPIs).forEach(API => {
           assert.equal(manifest.name, 'second')
           assert.equal(manifest.type, 'feed')
           assert.notEqual(manifest.accessController, null)
-          assert.equal(manifest.accessController.indexOf('/ipfs'), 0)
+          assert.ok(manifest.accessController.indexOf('/ipfs') === 0 || manifest.accessController.indexOf('\\ipfs') === 0)
```
Comment (haadcode) [inline test/create-open.test.js]: It'd be great if you could add an explicit test for testing `\` -> `/` conversion and to provide an example how the addresses and paths look like on Windows platforms. Not sure how exactly to test it, though. What do you think?

File: package.json (hunk 1 above)
Comment (durac) [reply package.json]: You are right, this line isn't really necessary. I'll delete it.

**Comment by BartKnucle** [comment]
Nice job, I'am able to use orbitDB on windows 10 now! Thx!

**Comment by JulianaDixon** [comment]
Hey! Thanks for this. I've run it on my computer, and three tests seem to have issues. Thoughts?

```sh
  1) orbit-db - Create & Open (js-ipfs)
       Create
         Success
           can pass local database directory as an option:
     Uncaught AssertionError [ERR_ASSERTION]: NotFoundError: Key not found in database [/orbitdb/QmQAAJKY6kn8Ac4cVFxDfvKwqMr38PHdsCBYX46XB1Cgt6/second/_manifest]
    at C:... == null
      at db.get (test\create-open.test.js:125:24)
      at node_modules\levelup\lib\levelup.js:168:14

  2) orbit-db - Create & Open (js-ipfs)
       Create
         Success
           Access Controller
             "before all" hook for "can pass local database directory as an option":
     Error: IO error: LockFile orbitdb\tests\create-open\QmQAAJKY6kn8Ac4cVFxDfvKwqMr38PHdsCBYX46XB1Cgt6\second/LOCK: The process cannot access the file because it is being used by another process.



  3) orbit-db - Create Custom Database Type (js-ipfs)
       "before all" hook:
     Error: EBUSY: resource busy or locked, unlink 'orbitdb\tests\create-open\QmQAAJKY6kn8Ac4cVFxDfvKwqMr38PHdsCBYX46XB1Cgt6\second\000005.ldb'
      at Object.unlinkSync (fs.js:949:3)
      at rimrafSync (node_modules\rimraf\rimraf.js:306:17)
      at node_modules\rimraf\rimraf.js:342:5
      at Array.forEach (<anonymous>)
      at rmkidsSync (node_modules\rimraf\rimraf.js:341:26)
      at rmdirSync (node_modules\rimraf\rimraf.js:334:7)
      at rimrafSync (node_modules\rimraf\rimraf.js:304:9)
      at node_modules\rimraf\rimraf.js:342:5
      at Array.forEach (<anonymous>)
      at rmkidsSync (node_modules\rimraf\rimraf.js:341:26)
      at rmdirSync (node_modules\rimraf\rimraf.js:334:7)
      at rimrafSync (node_modules\rimraf\rimraf.js:304:9)
      at node_modules\rimraf\rimraf.js:342:5
      at Array.forEach (<anonymous>)
      at rmkidsSync (node_modules\rimraf\rimraf.js:341:26)
      at rmdirSync (node_modules\rimraf\rimraf.js:334:7)
      at Function.rimrafSync [as sync] (node_modules\rimraf\rimraf.js:304:9)
      at Context.before (test\create-type.test.js:39:12)
```

**Comment by JulianaDixon** [comment]
Hmm. I had some issues running the examples.

``` 
$ npm run build:examples

> orbit-db@0.19.9 build:examples C:\Users\julia\Code\orbit-db
> webpack --config conf/webpack.example.config.js --sort-modules-by size && mkdir -p examples/browser/lib && cp node_modules/ipfs/dist/index.js examples/browser/lib/ipfs.js

Hash: f5801aa88f822e51bdef
Version: webpack 3.12.0
Time: 10074ms
                                               Asset     Size  Chunks                    Chunk Names
./examples/browser/browser-webpack-example/bundle.js  4.63 MB       0  [emitted]  [big]  main
 [744] (webpack)/buildin/amd-options.js 82 bytes {0} [built]
[1621] ./src/db-manifest.js 454 bytes {0} [built]
   [5] (webpack)/buildin/global.js 509 bytes {0} [built]
  [28] (webpack)/buildin/module.js 517 bytes {0} [built]
 [636] ./examples/browser/browser-webpack-example/index.js 755 bytes {0} [built]
[1618] ./src/ipfs-access-controller.js 1.06 kB {0} [built]
[1620] ./src/orbit-db-address.js 1.4 kB {0} [built]
[1622] ./src/exchange-heads.js 1.41 kB {0} [built]
[1619] ./src/access-controller.js 1.96 kB {0} [built]
[1628] ./examples/browser/example.js 8.95 kB {0} [built]
[1576] ./src/OrbitDB.js 13.4 kB {0} [built]
    + 1618 hidden modules
The syntax of the command is incorrect.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! orbit-db@0.19.9 build:examples: `webpack --config conf/webpack.example.config.js --sort-modules-by size && mkdir -p examples/browser/lib && cp node_modules/ipfs/dist/index.js examples/browser/lib/ipfs.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the orbit-db@0.19.9 build:examples script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\julia\AppData\Roaming\npm-cache\_logs\2018-12-18T17_36_38_642Z-debug.log

```

**Comment by RichardLitt** [comment]
Note: The above was me. I'm not sure what the problem was.

**Comment by HTUlv412** [comment]
Any information about this issue? we are waiting for the fix :)

**Comment by BartKnucle** [comment]
@HTUlv412 Fix is working :) Not merge yet but just modify the files according to the commit will solve the base58 error.

**Comment by RichardLitt** [comment]
I have a Windows computer to run this now; I'll be back in touch with it next week. This may not be the best way to manage our Windows environment maintenance, though; I'm not an expert at Windows, although I can run the tests and just generally see if it works. 

Windows is not supported yet, but if this PR works, then perhaps we can just merge it. I'm not sure if that is best practice or not. @aphelionz @haadcode How would you feel about merging this based on the positive reviews above?

**Comment by aphelionz** [comment]
I believe Windows compatibility was addressed here: https://github.com/orbitdb/orbit-db/pull/712

Closing. Please feel free to re-open if I'm not correct.

**Pull request closed without merging**